.github/actions/verify-token/src/predicate.ts

```diff
@@ -138,12 +137,25 @@ export function createPredicate(
   };
 
   const payload = github.context;
+  // NOTE: see example at https://github.com/slsa-framework/slsa/blob/main/docs/github-actions-workflow/examples/v0.1/example.json.
   const predicate: SLSAv1Predicate = {
     buildDefinition: {
       buildType: DELEGATOR_BUILD_TYPE,
       externalParameters: {
+        // Inputs to the TRW, which define the interface of the builder for the
+        // BYOB framework.
+        // TODO(#1575): mask inputs.
+        inputs: rawTokenObj.tool.inputs,
+        // Variables are always empty for BYOB / builders.
+        // TODO(#1555): add support for generators.
+        vars: {},
         // NOTE: This is equivalent to the v0.2 entryPoint.
-        workflowPath: currentRun.path,
+        workflow: {
+          ref: env.GITHUB_REF,
+          repository: env.GITHUB_REPOSITORY,
```

Comment:
I guess this should be the same but should this info not be taken from the token? Maybe from `rawTokenObj.github.env.XXX`

Reply:
Correct, these values should be the same, since they are part of the same run.

Reply:
NOTE: we use these values in https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/actions/verify-token/src/predicate.ts#L128-L131 to populate the source as well.

Reply:
Yeah,
https://github.com/slsa-framework/slsa-github-generator/blob/main/.github/actions/verify-token/src/main.ts#L256-L266

Reply:
yes, it should, see function `validateGitHubFields` in main.ts

Reply:
Shall we add a comment in the code?

Reply:
I think what we should be really doing is validating that the data on the token object is correct prior to this, and using data on the token rather than data that may or may not be validated. It *should* be the same run but I think it could be written in a safer way.

Not a major issue right now though so I'll send a follow up PR myself.